### PR TITLE
[15.0] add base_multicompany_reporting_currency

### PR DESCRIPTION
**Pull request by yankinmax**
This module is needed to set Group currency that will be applied to all companies.
group_curr

### CODE REVIEW

**Comment by rousseldenis** [comment]
@yankinmax I don't see where that parameter is applied to all companies. Do I miss something ?

File: base_group_currency/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    company_group_currency_id = fields.Many2one(
+        "res.currency",
+        string="Group Currency",
+        config_parameter="base_group_currency.group_currency_id",
```
Comment (yankinmax) [inline base_group_currency/models/res_config_settings.py]: @rousseldenis here is it

File: base_group_currency/data/default_group_currency.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<odoo>
+    <record id="group_currency_id" model="ir.config_parameter">
+        <field name="key">base_group_currency.group_currency_id</field>
+        <field name="value" ref="base.EUR" />
```
Comment (yankinmax) [inline base_group_currency/data/default_group_currency.xml]: @rousseldenis and this is default value

**Comment by rousseldenis** [comment]
Ok, what I wanted to say is following your description, we could think that when creating a new company, the default currency will be that one.

**Comment by yankinmax** [comment]
@rousseldenis I'm not good at writing descriptions)))
How would you rephrase it to be more clear? Pls, I really appreciate your help.

**Comment by rousseldenis** [comment]
> @rousseldenis I'm not good at writing descriptions))) How would you rephrase it to be more clear? Pls, I really appreciate your help.

I will try but not too much time.

First, you need to identify the functional needs for this. For what will it be used?

File: base_group_currency/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Base Group Currency",
+    "summary": "Adds the possibility to specify Group currency",
+    "version": "15.0.1.0.1",
```
Comment (i-vyshnevska) [inline base_group_currency/__manifest__.py]:
```suggestion
    "version": "15.0.1.0.0",
```

**Review by i-vyshnevska — APPROVED** [review]

**Comment by i-vyshnevska** [comment]
@rousseldenis  this currency field will be used in other places to compare amounts in different companies and documents, by standalone this addon doesn't bring functionality only in dependency with other development

**Comment by rousseldenis** [comment]
> @rousseldenis this currency field will be used in other places to compare amounts in different companies and documents, by standalone this addon doesn't bring functionality only in dependency with other development

This should be added in README and IMHO, in configuration you need to add a help too. I think the 'Group' term is maybe inappropriate. Maybe 'Inter-Company'

File: base_group_currency/__manifest__.py (hunk 3 above)
Comment (yankinmax) [reply base_group_currency/__manifest__.py]: bad copy-paste, thanks

**Review by ajaniszewska-dev — APPROVED** [review]

**Review by gurneyalex — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]
